holoviews/core/dimension.py

```diff
@@ -767,8 +770,9 @@ def select(self, selection_specs=None, **kwargs):
 
         # Apply selection to self
         if local_kwargs and matches:
-            ndims = (len(self.dimensions()) if any(d in self.vdims for d in kwargs)
-                     else self.ndims)
+            ndims = self.ndims
+            if any(d in self.vdims for d in kwargs):
+                ndims = len(self.kdims+self.vdims)
```

Comment:
No, because it can include the deep dimensions as well which should be processed below.